libs/pilight/protocols/433.92/hideki.c

```diff
@@ -0,0 +1,509 @@
+	struct settings_t *tmp = NULL;
+	while(settings) {
+		tmp = settings;
+		settings = settings->next;
+		FREE(tmp);
+	}
+	if(settings != NULL) {
+		FREE(settings);
+	}
+}
+
+#if !defined(MODULE) && !defined(_WIN32)
+__attribute__((weak))
+#endif
+void hidekiInit(void) {
+	protocol_register(&hideki);
+	protocol_set_id(hideki, "hideki");
+	protocol_device_add(hideki, "TS04", "HIDEKI TS04 sensor");
+	protocol_device_add(hideki, "Hideki-Wind", "HIDEKI Wind sensor");
+	protocol_device_add(hideki, "Hideki-Temp", "HIDEKI Temperature sensor");
+	protocol_device_add(hideki, "Hideki-Rain", "HIDEKI Rain sensor");
+	hideki->devtype = WEATHER;
+	hideki->hwtype = RF433;
+	hideki->maxgaplen = MAX_PULSE_LENGTH*PULSE_DIV;
+	hideki->mingaplen = MIN_PULSE_LENGTH*PULSE_DIV;
+	hideki->minrawlen = MIN_RAW_LENGTH;
+	hideki->maxrawlen = MAX_RAW_LENGTH;
+
+	// device values
+	options_add(&hideki->options, "t", "temperature", OPTION_HAS_VALUE, DEVICES_VALUE, JSON_NUMBER, NULL, "^[0-9]{1,3}$");
+	options_add(&hideki->options, "h", "humidity", OPTION_HAS_VALUE, DEVICES_VALUE, JSON_NUMBER, NULL, "^[0-9]{1,5}$");
+	options_add(&hideki->options, "w", "wind", OPTION_HAS_VALUE, DEVICES_VALUE, JSON_NUMBER, NULL, "^[0-9]{1,5}$");
+	options_add(&hideki->options, "d", "wind_direction", OPTION_HAS_VALUE, DEVICES_VALUE, JSON_NUMBER, NULL, "^[0-9]{1,5}$");
+	options_add(&hideki->options, "r", "rain", OPTION_HAS_VALUE, DEVICES_VALUE, JSON_NUMBER, NULL, "^[0-9]{1,5}$");
+	options_add(&hideki->options, "b", "battery", OPTION_HAS_VALUE, DEVICES_VALUE, JSON_NUMBER, NULL, "^[01]$");
+	options_add(&hideki->options, "o", "rc", OPTION_HAS_VALUE, DEVICES_VALUE, JSON_NUMBER, NULL, "[0-9]"); // rolling code, might be used as ID, if constant
+	// device IDs
+#if 0
+	options_add(&hideki->options, "i", "id", OPTION_HAS_VALUE, DEVICES_ID, JSON_NUMBER, NULL, "[0-9]");
+#endif
+	options_add(&hideki->options, "c", "channel", OPTION_HAS_VALUE, DEVICES_ID, JSON_NUMBER, NULL, "[0-9]");
+	// device settings:
+	options_add(&hideki->options, "0", "temperature-offset", OPTION_HAS_VALUE, DEVICES_SETTING, JSON_NUMBER, (void *)0, "[0-9]");
+	options_add(&hideki->options, "0", "humidity-offset", OPTION_HAS_VALUE, DEVICES_SETTING, JSON_NUMBER, (void *)0, "[0-9]");
+	options_add(&hideki->options, "0", "wind-direction-offset", OPTION_HAS_VALUE, DEVICES_SETTING, JSON_NUMBER, (void *)0, "[0-9]");
+	options_add(&hideki->options, "0", "wind-factor", OPTION_HAS_VALUE, DEVICES_SETTING, JSON_NUMBER, (void *)1, "^[0-9]{1,5}$");
+	options_add(&hideki->options, "0", "rain-factor", OPTION_HAS_VALUE, DEVICES_SETTING, JSON_NUMBER, (void *)1, "^[0-9]{1,5}$");
+	/* tests have shown that the transmitted temperature value might be damages in case if battery=0.
+	 * As workaround, the temperature will not be output in case of battery=0: This behavior can be activated by this setting */
+	options_add(&hideki->options, "0", "ignore-temperature-on-battery-low", OPTION_NO_VALUE, DEVICES_SETTING, JSON_NUMBER, NULL, NULL);
```

Comment:
Why not make this the default?

Reply:
I thought about this. For me it's a kind of workaround and I don't know the reason of this behaviour. I could test the sensor only with one single device. Maybe it's specific for my device or the location of the sensor. As long as I have no indication that this is a general problem of this sensors I prefer to leave it optional and switched off per default.

Reply:
Was this protocol already discussed on the forum. Maybe other users can tune in?

Reply:
the only location in the forum I remember is this one:
https://forum.pilight.org/showthread.php?tid=3421